### PR TITLE
Redirects preserve query parameters.

### PR DESCRIPTION
**Pull request by mbleigh**
This updates redirects to properly match and redirect when a query string is present, preserving the query string. It explicitly **does not**:

1. Allow matching based on query strings
2. Allow captures to be inserted into query strings

### CODE REVIEW

File: lib/middleware/redirects.js [hunk 1]
```diff
@@ -54,15 +61,15 @@ Redirect.prototype.test = function(url) {
 
       return {
         type: this.type,
-        destination: dest
+        destination: dest + qs
```
Comment (drtriumph) [inline lib/middleware/redirects.js]: I think it's possible for destinations to contain a query string - do we want to check and possibly merge query strings (overwriting? adding? if we get a collision)

File: lib/middleware/redirects.js [hunk 2]
```diff
@@ -34,6 +34,13 @@ var Redirect = function(source, destination, type) {
 };
 
 Redirect.prototype.test = function(url) {
+  var qs = '';
```
Comment (drtriumph) [inline lib/middleware/redirects.js]: Should we explicitly remove the query string in all places we match (rewrites, headers etc.)?

Comment (mbleigh) [inline lib/middleware/redirects.js]: We don't need to - it's `pathToRegexp` that falls apart with a query string, everywhere else we use `minimatch` which handles these things just fine:

    minimatch('/foo/bar?baz=qux', '/foo/*'); // => true

File: lib/middleware/redirects.js (hunk 1 above)
Comment (mbleigh) [inline lib/middleware/redirects.js]: Let's keep it simple -- I'm now appending them, including a dup if there's a dup between the supplied qs and the destination qs.

**Pull request merged**